src/main/java/ust/tad/ansiblempsplugin/models/tadm/Property.java

```diff
@@ -4,156 +4,181 @@
-        if(isValueMatchingType(this.type, value)) {
-            this.value = value;
-        } else {
-            throw new InvalidPropertyValueException(String.format(INVALIDPROPERTYVALUEEXCEPTIONMESSAGE, value.toString(), value.getClass().toString(), this.type.toString()));
-        }
-    }
-    
-    public Confidence getConfidence() {
-        return this.confidence;
-    }
-
-    public void setConfidence(Confidence confidence) {
-        this.confidence = confidence;
-    }
-    
-    public Property key(String key) {
-        setKey(key);
-        return this;
-    }
-
-    public Property type(PropertyType type) throws InvalidPropertyValueException {
-        setType(type);
-        return this;
-    }
-
-    public Property required(boolean required) {
-        setRequired(required);
-        return this;
-    }
-
-    public Property value(Object value) throws InvalidPropertyValueException {
-        setValue(value);
-        return this;
-    }
-
-    public Property confidence(Confidence confidence) {
-        setConfidence(confidence);
-        return this;
-    }
-
-    @Override
-    public boolean equals(Object o) {
-        if (o == this)
-            return true;
-        if (!(o instanceof Property)) {
-            return false;
-        }
-        Property property = (Property) o;
-        return Objects.equals(key, property.key) && Objects.equals(type, property.type) && required == property.required && Objects.equals(value, property.value) && Objects.equals(confidence, property.confidence);
-    }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(key, type, required, value, confidence);
-    }
-
-    @Override
-    public String toString() {
-        return "{" +
-            " key='" + getKey() + "'" +
-            ", type='" + getType() + "'" +
-            ", required='" + isRequired() + "'" +
-            ", value='" + getValue() + "'" +
-            ", confidence='" + getConfidence() + "'" +
-            "}";
-    }
-
-    private boolean isValueMatchingType(PropertyType type, Object value) {
-        if(value == null || type == null) {
-            return true;
-        } else {
-            switch(type) {
-                case BOOLEAN:
-                    return value.getClass() == Boolean.class;
-                case DOUBLE:
-                    return value.getClass() == Double.class || value.getClass() == Float.class;
-                case INTEGER:
-                    return value.getClass() == Integer.class || value.getClass() == Long.class;
-                case STRING:
-                    return value.getClass() == String.class;
-                default:
-                    return false;
-            }
-        }        
-    }
-    
-
+  private String key;
+
+  private PropertyType type;
+
+  private boolean required;
+
+  private Object value;
+
+  private Confidence confidence = Confidence.CONFIRMED;
+
+  private static final String INVALIDPROPERTYVALUEEXCEPTIONMESSAGE =
```

Comment:
The name is really long and confusing. I get why it's called like that but wouldn't `EXCEPTIONTEMPLATE` also work? That way the name is shorter and still carries the same meaning.

Reply:
idk this is copied from Marcel, I would like to keep it consistent...